WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+<?php
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ */
+
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * Provides a bootstrap for the sniffs, to reduce code duplication.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Juliette Reinders Folmer <wpplugins_nospam@adviesenzo.nl>
+ * @author    khacoder
+ * @author    Ulrich Pogson <ulrich@pogson.ch>
+ */
+abstract class WordPress_AbstractThemeSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Array holding the once through WordPress Theme file checks.
+	 *
+	 * @var array
+	 */
+	private static $sniff_helper;
+
+	/**
+	 * Initial setup.
+	 */
+	public function __construct() {
+		// do a once through pass and get the files.
+		if ( empty( self::$sniff_helper ) || ! is_array( self::$sniff_helper ) || false === self::$sniff_helper ) {
+			$files = $this->get_files();
+			if ( isset( $files[0] ) ) {
```

Comment:
This will fail depending on the method the command line is run. e.g. `phpcs --standard=WordPress-Theme functions.php`, `phpcs --standard=WordPress-Theme *`, `phpcs --standard=WordPress-Theme .` or `phpcs --standard=WordPress-Theme wp-content/themes/_s`

This should help to support the other methods. https://gist.github.com/grappler/868213a2a0ed4a700173a7f3105ecdce#file-abstractthemesniff-php-L40